test/stdcpp/src/allocator_test.d

```diff
@@ -0,0 +1,14 @@
+import core.stdcpp.allocator;
+
+extern(C++) struct Mystruct { int a; }
+
+alias AliasSeq(T...) = T;
+unittest
+{
+    static foreach(T; AliasSeq!(int, float, Mystruct))
```

Comment:
Should there also be a test for:
- `extern(C++) class`
- `Object` / `extern(D) class`

Reply:
`std::allocator` doesn't work with D classes... it would just allocate a class pointer. should I wrap it up so that it does?

Reply:
Actually, no, it's impossible to instantiate with a class. D has no syntax to refer to the classes actual type, only the pointer to it.

Reply:
It assert's a helpful message.